### PR TITLE
Chore(ci): Introduce GitHub Action for Storybook Deploy to GH Pages

### PR DESCRIPTION
**Pull request by literat**

### CODE REVIEW

File: .editorconfig
Comment (adamkudrna) [inline .editorconfig:23]: This rule is not necessary then, there are 2 spaces by default. Therefore the entire [*.yml] and also [package.json] sections can be removed.

File: .editorconfig
Comment (adamkudrna) [inline .editorconfig:20]: Once we are around, I believe using whitespace at the end of MD lines is tricky and easy to miss. It's better to avoid using this MD feature and prefer `<br />` or new paragraph instead. (It's rarely needed anyway, at least according to my experience.)

If you agree, we could end up with a neat and simple default `.editorconfig` 🙂.

File: .github/workflows/storybook.yml
Comment (adamkudrna) [inline .github/workflows/storybook.yml:3]: Let's leave it to all pushes for now, but we will probably prefer deploying docs only upon release so the docs always corresponds to the latest version published.

File: .github/workflows/storybook.yml
Comment (adamkudrna) [inline .github/workflows/storybook.yml:7]: I'd rather stick to the same Ubuntu version in all actions. LTS is more predictable and maintainable than `latest`, so I suggest to put `ubuntu-20.04` here.

File: .github/workflows/storybook.yml
Comment (adamkudrna) [inline .github/workflows/storybook.yml:10]: `@v2.3.1` is OK, but maybe `@2` is just fine? (See other GH workflows.)

The same goes for other actions in all workflows — unless we really want to use a specific version. Either is fine but it should be the same way everywhere.

**Review by adamkudrna — CHANGES_REQUESTED** [review]
Cool! Are we able to test the deploy before merge? 🙂

**Comment by literat** [comment]
Testing deploy: https://lmc-eu.github.io/spirit-design-system

File: .github/workflows/storybook.yml
Comment (literat) [reply .github/workflows/storybook.yml]: For testing deploy I switched it to all pushes for now. However I want to deploy storybook only on push to main branch and also `paths` option sounds good to me.

File: .github/workflows/storybook.yml
Comment (literat) [reply .github/workflows/storybook.yml]: True, I have changed it already.

File: .github/workflows/storybook.yml
Comment (literat) [reply .github/workflows/storybook.yml]: For checkout us `@v2` just fine. But for `JamesIves/github-pages-deploy-action` the `@4` does not work. So I must stay with `@4.1.4`.

File: .editorconfig
Comment (literat) [reply .editorconfig]: Ok, removed.

File: .github/workflows/storybook.yml
Comment (adamkudrna) [reply .github/workflows/storybook.yml]: I'm a little bit confused with versions you are mentioning, but I believe it's a typo and I get the point 🙂.

File: .github/workflows/storybook.yml
Comment (adamkudrna) [reply .github/workflows/storybook.yml]: OK, let's have it this way now.

File: .editorconfig
Comment (adamkudrna) [reply .editorconfig]: Maybe let's skip it and focus on default `.editorconfig` in https://github.com/lmc-eu/code-quality-tools.

File: .github/workflows/storybook.yml
Comment (adamkudrna) [inline .github/workflows/storybook.yml:4]: I still believe the docs should be deployed only upon release — unless we also want a WIP docs version (corresponding to the `main` branch) which might be useful as well. As a developer, I want to see the docs of the _latest version released_ so I know what I can use. But again, this can be discussed at any point in the future.

**Review by adamkudrna — APPROVED** [review]

File: .github/workflows/storybook.yml
Comment (adamkudrna) [reply .github/workflows/storybook.yml:4]: Changed to pushes only to the `main` branch.

**Pull request merged**
